File: Readability/Readability.c
#include <stdio.h>
#include <cs50.c>
#include <ctype.h>
#include <math.h>
#include <string.h>

int main()
{
    string text = get_string("Text:"); //prompt user for the text

    int letters = 0 ;        //define all the vars
    int sents = 0 ;
    int words = 1 ;          //words = 1 because we have a word in the first

    for (int i = 0 ; i < strlen(text) ; i ++)  //make a loop to check all the strings using strlen which calculate the text lenghth
    {
        if ((text[i] != ' ') && (ispunct(text[i]) == false))
        {
            letters ++ ;
        }

        if ((text[i] == '.') || (text[i] == '!') || (text[i] == '?'))
        {
            sents ++ ;
        }

        if (((text[i] == ' ') && (text[ i + 1] != ' ')) || ((text[i] == 39) && (text[i + 1] == 'v')))
        {
            words ++ ;
        }
    }

    float l = (letters * 100) / words ; //l is the average letters every 100 words
    float s = (sents * 100) / words ; // s is the average senteces every 100 words

    float index = 0.0588 * l - 0.296 * s - 15.8 ; // then apply the algorithm

    int grade = round(index);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

File: PSet6/Readability.py
from cs50 import get_string

text = get_string("Text :")  # asking for text
array = text.split(" ")  # putting it in an array

words = 0
letters = 0
sentences = 0

for i in array:
    if len(i) != 1 or i == "I" or i.lower() != "the":  # counting the words
        words += 1

    for j in i:  # counting the letters by (isalpha) function
        if j.isalpha():
            letters += 1

    if ("." in i) or ("!" in i) or ("?" in i):  # counting senteces
        sentences += 1

L = letters / words * 100
S = sentences / words * 100
grade = round(0.0588 * L - 0.296 * S - 15.8)  # substitute in the formula

if grade < 16 and grade > 0:  # printing
    print("Grade " + str(grade))
elif grade < 1:
    print("Before Grade 1")
else:
    print("Grade 16+")
